### DISCUSSION TITLE
Telescope find only searches for file relative to a particular directory.

### DISCUSSION THREAD

**Discussion by ivange94** [LazyVim/LazyVim#560] [General]
Recently I've been having an issue with telescope find. Here's my config https://github.com/ivange94/nvim. For some reason I only experience this with one project and with certain files. This is the project I'm talking about https://github.com/iotaledger/wasp. In that project, there's a file called `magic.md` in the `documentation` folder. If I do `<leader> ff` to search for this file, my workspace directory is changed to the documentation folder. So subsequent searches of files that fall outside of that folder but still within the project don't work.

**Comment by ivange94** [comment]
With more investigation, I think I figured out what's causing the problem but I don't know how to fix it. I think it's how my config detects what a workspace is. The `documentation` has a `package.json` file so it's treating that folder like a separate workspace. If I delete that file, I no longer have this problem. But deleting that file is not an acceptable solution.

**Reply by robsonpeixoto** [reply]
@ivange94 I'm neovim/LazyVim noob and I didn't know how to solve it. But read it: https://github.com/LazyVim/LazyVim/blob/7ce1b1519a45f48aa4b5f35b7b7a886b42fe56d8/lua/lazyvim/util/init.lua#L43-L82

**Reply by briandipalma** [reply]
You need to learn the difference between `<leader>ff` and `<leader>fF` or `cwd` and `root` directory. I had the same problem as in my previous config I didn't distinguish but once you get used to it I think it makes sense. Although I would prefer to switch `f` and `F` as I more often perform cwd searches.

**Reply by danielo515** [reply]
@briandipalma do you mind explaining the difference in simple terms? I can't understand it.

EDIT: I just found this, is it correct?
> In Neovim, the "cwd" (current working directory) is the directory from which the editor was launched, and is the default location for opening files and saving changes. The "root directory" is the top-level directory of the current project, as determined by the presence of a configuration file (such as a `.git` directory). The root directory is used by some plugins and commands to determine the context of the project, and can be set manually using the `:cd` command.

**Reply by briandipalma** [reply]
> EDIT: I just found this, is it correct?

Yes, the root directory changes dynamically based on the buffer you have focused. If you are working in a monorepo with many npm packages each one would be a root directory so when you have a `.ts` file open from one of them and perform a root directory search then the search is limited to that npm package and won't find files in the rest of the monorepo.

**Reply by StevenXL** [reply]
Thank you! This is great. I wish `root` mean the top-level and `cwd` meant the nested level, but I can live with this. Just gotta keep in mind that `cwd` => static dir from where `nvim` was lauchned. `root` => dynamic directory.

**Comment by neonvoidx** [comment]
> > EDIT: I just found this, is it correct?
> 
> Yes, the root directory changes dynamically based on the buffer you have focused. If you are working in a monorepo with many npm packages each one would be a root directory so when you have a `.ts` file open from one of them and perform a root directory search then the search is limited to that npm package and won't find files in the rest of the monorepo.

Ok how tf do we fix this if we want, if im in a folder that has 3 subfolders and each subfolder is a git repo, how do i make it so that things like telescope or neotree always open/search at the main folder level? I can't seem to get it to work

**Reply by dpetka2001** [reply]
See answer [here](https://github.com/LazyVim/LazyVim/discussions/1714#discussioncomment-7281079)

**Reply by briandipalma** [reply]
I don't use neo-tree any longer I prefer the mini.files extra. But as for telescope if I want to perform a local to the buffers project file search I hit `<leader>ff` (or `<leader><leader>`) and if I want to perform a cwd search I hit `<leader>fF`. That pattern is repeated for the other searches, lower case project local and upper case cwd global.

Although with proper code structure you should be relying on the lower case file/search operations more than the upper case ones. If you give it some time you will get used to it and it's well worth it. Nested git repos sounds like a mess TBH.
